### PR TITLE
feat: Add contributor's insight on the workspace.

### LINKED ISSUE #3948: Feature: show where to find the contributors' insight page
### Suggested solution

While looking at the project, I didn't know you could create a contributors' insight from workspaces, so I'm proposing we add that to the home page for workspace somewhere here rather than the side nav bar that's hidden:

image

or the side nav should be open by default on desktop

### PR DESCRIPTION
**Pull request by oyezaheer**
Fixes #3948

Screenshot 2024-08-28 234402

### CODE REVIEW

**Comment by jpmcb** [comment]
Thanks for this contribution @oyezaheer - this looks to contain changes from one of your previous PRs. Please get the latest changes from the `beta` branch and clean this up and we'll move this forward.

Also looks like the Playwright tests are failing:

```

./pages/workspaces/[workspaceId]/activity.tsx:133:12
Type error: Type '{ workspace: Workspace; }' is missing the following properties from type 'WorkspaceHeaderProps': title, url

  131 |       >
  132 |         <div className="px-4 py-8 lg:px-16 lg:py-12">
> 133 |           <WorkspaceHeader workspace={workspace} />
      |            ^
  134 |           <div className="grid sm:flex gap-4 pt-3 border-b">
  135 |             <WorkspacesTabList workspaceId={workspace.id} selectedTab={"activity"} />
  136 |           </div>
```

**Comment by nickytonline** [comment]
> Thanks for this contribution @oyezaheer - this looks to contain changes from one of your previous PRs. Please get the latest changes from the `beta` branch and clean this up and we'll move this forward.
> 
> Also looks like the build and Playwright tests are failing. Please make sure everything builds before requesting review.
> 
> ```
> 
> ./pages/workspaces/[workspaceId]/activity.tsx:133:12
> Type error: Type '{ workspace: Workspace; }' is missing the following properties from type 'WorkspaceHeaderProps': title, url
> 
>   131 |       >
>   132 |         <div className="px-4 py-8 lg:px-16 lg:py-12">
> > 133 |           <WorkspaceHeader workspace={workspace} />
>       |            ^
>   134 |           <div className="grid sm:flex gap-4 pt-3 border-b">
>   135 |             <WorkspacesTabList workspaceId={workspace.id} selectedTab={"activity"} />
>   136 |           </div>
> ```

I updated the branch and the changes from the previous PRs are gone.

File: components/Workspaces/WorkspaceHeader.tsx [hunk 1]
```diff
@@ -47,6 +49,17 @@ export const WorkspaceHeader = ({ workspace, children }: WorkspaceHeaderProps) =
       </h1>
       <div className="flex gap-2 justify-end">
```
Comment (nickytonline) [inline components/Workspaces/WorkspaceHeader.tsx:50]: refactor (blocking): On smaller screens, the feature is not responsive. It causes other buttons to go off screen and will cause the layout for any page using the WorkspacesHeader to break.

CleanShot 2024-08-29 at 08 22 30

Adding flex wrap should sort it out. 👇

CleanShot 2024-08-29 at 08 27 52

```suggestion
      <div className="flex gap-2 justify-end flex-wrap">
```

**Review by nickytonline — CHANGES_REQUESTED** [review]

**Review by nickytonline — APPROVED** [review]
Thanks @oyezaheer! :shipit:

**Review by brandonroberts — APPROVED** [review]
🍕

**Pull request merged**